ow_sim_tests/launch/empty_world.launch

```diff
@@ -0,0 +1,57 @@
+<?xml version="1.0"?>
+<launch>
+
+  <!-- these are the arguments you can pass this launch file, for example paused:=true -->
+  <arg name="paused" default="false"/>
+  <arg name="use_sim_time" default="true"/>
+  <arg name="extra_gazebo_args" default=""/>
+  <arg name="gui" default="true"/>
+  <arg name="recording" default="false"/>
+  <!-- Note that 'headless' is currently non-functional.  See gazebo_ros_pkgs issue #491 (-r arg does not disable
+       rendering, but instead enables recording). The arg definition has been left here to prevent breaking downstream
+       launch files, but it does nothing. -->
+  <arg name="headless" default="false"/>
+  <arg name="debug" default="false"/>
+  <arg name="physics" default="ode"/>
+  <arg name="verbose" default="false"/>
+  <arg name="output" default="screen"/>
+  <arg name="world_name" default="worlds/empty.world"/> <!-- Note: the world_name is with respect to GAZEBO_RESOURCE_PATH environmental variable -->
+  <arg name="respawn_gazebo" default="false"/>
+  <arg name="use_clock_frequency" default="false"/>
+  <arg name="pub_clock_frequency" default="100"/>
+  <arg name="enable_ros_network" default="true" />
+  <arg name="server_required" default="false"/>
+  <arg name="gui_required" default="false"/>
+
+  <!-- set use_sim_time flag -->
+  <param name="/use_sim_time" value="$(arg use_sim_time)"/>
+
+  <!-- set command arguments -->
+  <arg unless="$(arg paused)" name="command_arg1" value=""/>
+  <arg     if="$(arg paused)" name="command_arg1" value="-u"/>
+  <arg unless="$(arg recording)" name="command_arg2" value=""/>
+  <arg     if="$(arg recording)" name="command_arg2" value="-r"/>
+  <arg unless="$(arg verbose)" name="command_arg3" value=""/>
+  <arg     if="$(arg verbose)" name="command_arg3" value="--verbose"/>
+  <arg unless="$(arg debug)" name="script_type" value="gzserver"/>
+  <arg     if="$(arg debug)" name="script_type" value="debug"/>
+
+  <!-- start gazebo server-->
+  <group if="$(arg use_clock_frequency)">
+    <param name="gazebo/pub_clock_frequency" value="$(arg pub_clock_frequency)" />
+  </group>
+  <group>
+    <param name="gazebo/enable_ros_network" value="$(arg enable_ros_network)" />
+  </group>
+  
+  <node name="gazebo" pkg="gazebo_ros" type="$(arg script_type)" respawn="$(arg respawn_gazebo)" output="$(arg output)"
+	args="$(arg command_arg1) $(arg command_arg2) $(arg command_arg3) -e $(arg physics) $(arg extra_gazebo_args) $(arg world_name)"
+  required="$(arg server_required)" />
+
+  <!-- start gazebo client -->
+  <group if="$(arg gui)">
+    <node name="gazebo_gui" pkg="gazebo_ros" type="gzclient" respawn="false" output="$(arg output)" args="$(arg command_arg3) -g libow_fps_monitor.so"
```

Comment:
You have near duplicates of empty_world.launch, common.launch, and atacama_y1a.launch. It would be much cleaner to use the original files and override the parts that need to be different.

I could be wrong, but it looks like you needed to do all this to inject "-g libow_fps_monitor.so." There should be a way to inject that without duplicating empty_worlds.launch. I think I did that in the past with the extra_gazebo_args argument, but I'm not certain.

Reply:
Oops. It looks like someone asked about this already. But I still think there is a way to inject the plugin without copying the whole file.

Reply:
This was the only approach I found to this issue. I have had developed this plugin almost a year ago and back then this was the only viable solution. I wish there is a way to copy `empty_world.launch` as I hate having duplicates myself (so much).

Reply:
Terry, I created a new jira ticket https://babelfish.arc.nasa.gov/jira/browse/OCEANWATER-741 to take another look into the issue. I will start by checking with OSRF people during our meeting today.

Reply:
I have created another ticket https://babelfish.arc.nasa.gov/jira/browse/OCEANWATER-742 to refactor `common.launch` to also creating replicas in **ow_sim_tests**

Reply:
Just saw your comment. I'm able to inject your plugin and pass both tests using `<arg name="extra_gazebo_args" value="-g libow_fps_monitor.so joint_states:=/_original/joint_states"/>`.

Strangely, order matters. So you can't do `<arg name="extra_gazebo_args" value="joint_states:=/_original/joint_states" -g libow_fps_monitor.so/>`

Do you want to fix this up now or still do it as a separate ticket?

Reply:
I would say to address that in a separate ticket. The reason if this works then I want to refactor certain files.